src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/GetChangeFeedV33.sql

```diff
@@ -0,0 +1,48 @@
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     GetChangeFeed33
+--
+-- FIRST SCHEMA VERSION
+--     33
+--
+-- DESCRIPTION
+--     Gets a stream of dicom changes (instance adds and deletes)
+--
+-- PARAMETERS
+--     @limit
+--         * Max rows to return
+--     @offet
+--         * Rows to skip
+--     @startTime
+--         * Optional inclusive timestamp start
+--     @endTime
+--         * Optional exclusive timestamp end
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.GetChangeFeedV33 (
+    @startTime DATETIMEOFFSET(7),
+    @endTime   DATETIMEOFFSET(7),
+    @limit     INT,
+    @offset    BIGINT)
+AS
+BEGIN
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+
+    SELECT
+        Sequence,
+        Timestamp,
+        Action,
+        PartitionName,
+        StudyInstanceUid,
+        SeriesInstanceUid,
+        SopInstanceUid,
+        OriginalWatermark,
+        CurrentWatermark
+    FROM dbo.ChangeFeed c
+    INNER JOIN dbo.Partition p
+    ON p.PartitionKey = c.PartitionKey
+    WHERE c.Timestamp >= @startTime AND c.Timestamp <@endTime
```

Comment:
I like that the single prc can server both the old and new behavior.

But this SQL is not performant. There is no index on Timestamp, so this will a index scan on cluster index.
Even with the new index, we should look at the query plan and make sure only the limit rows are read when starttime and endtime is infinite

Reply:
I've altered the clustered index to include an ascending order on `Timestamp`. This is the new query plan for fetching change feed:

<img width="383" alt="image" src="https://user-images.githubusercontent.com/2955179/236285056-01fffe8e-2557-4ae1-95b0-f70497c97e4c.png">

Reply:
While the previous v1 latest entry sproc has a decent query plan that scans the new nonclustered index for a single row (since it's ordered), I think the other get change feed sproc query plan can be improved.

### GetChangeFeedLatestV6
Seems ok

<img width="588" alt="image" src="https://user-images.githubusercontent.com/2955179/236331429-12f1cdd7-f4c2-4071-b941-a052a69067b5.png">

### GetChangeFeedV6
Seems less than ideal... with a lot of focus on grabbing the changes per partition:

<img width="615" alt="image" src="https://user-images.githubusercontent.com/2955179/236338828-6d21ceb6-3ee3-49f6-a473-65422295a6f4.png">

Naively I would have thought the query plan for `GetChangeFeedV6` could be the same as `GetChangeFeedLatestV6` except that the top left `Index Scan` is replaced with an `Index Seek` + the sequence predicate